lib/compat/wordpress-6.6/class-gutenberg-rest-templates-controller-6-6.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Templates_Controller_6_6 class
+ *
+ * @package    gutenberg
+ */
+
+/**
+ * Gutenberg_REST_Templates_Controller_6_6 class
+ *
+ * Templates and template parts currently only allow access to administrators with the
+ * `edit_theme_options` capability. In order to allow other roles to also view the templates,
+ * we need to override the permissions check for the REST API endpoints.
+ */
+class Gutenberg_REST_Templates_Controller_6_6 extends Gutenberg_REST_Templates_Controller_6_4 {
+
+	/**
+	 * Checks if a given request has access to read templates.
+	 *
+	 * @since 6.6
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( current_user_can( 'edit_posts' ) ) {
+			return true;
+		}
+		foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
+			if ( current_user_can( $post_type->cap->edit_posts ) ) {
+				return true;
+			}
+		}
```

Comment:
This change appears to have caused an unintended regression.

Currently, Gutenberg uses the following code to determine whether the site editor can be accessed, mainly using the command palette, and switches the link destination based on the result.

```javascript
wp.data.select('core').canUser( 'read', 'templates' )
```

With this PR, this code now returns `true` even for non-administrator users, which causes links that they are not allowed to access to be displayed in commands, etc.

See #61419